Tests/MustacheTests/TemplateParserTests.swift

```diff
@@ -58,12 +58,42 @@ final class TemplateParserTests: XCTestCase {
     }
 }
 
+#if compiler(>=6.0)
+extension MustacheTemplate: @retroactive Equatable {
+    public static func == (lhs: MustacheTemplate, rhs: MustacheTemplate) -> Bool {
+        lhs.tokens == rhs.tokens
+    }
+}
```

Comment:
It would be nicer to put then`==` implementation in a singular place, and then add the conformance separately under compiler directive. Also, you can use `#if hasFeature(RetroactiveAttribute)`

Reply:
I was aware of `#if hasFeature(RetroactiveAttribute)` as I originally suggested it to Adam in a PR, but In this PR, i was going based on https://github.com/apple/swift-nio/pull/2719 which mentions using `#if compiler()`.

However, after double-checking things again myself, I think `#if hasFeature(RetroactiveAttribute)` is still pretty much correct, and more clear than `#if compiler()`. So I'll change things to use that instead.